YellowPages/firebase_functions/functions/src/email.ts

```diff
@@ -2,6 +2,24 @@ const mailgun = require('mailgun-js');
 const { getEmailSettings } = require('./firebase');
 // const { checkRecaptcha } = require('./helpers');
 
+const getDoubleOptIn = (packet, emailSettings) => {
+    const { emailRecepient } = emailSettings;
+    const { newsletter } = packet;
+    if (newsletter.toString() === 'true') {
+        //Add to pending list and send out confirmation email 
+        axios.post('https://newsletter-api.iota.org/api/signup',
```

Comment:
This is a side-effect. How does `getDoubleOptIn` know that it succeeded? Using a promise with a resolve() / reject() would allow the process to self-mediate and make the side-effect at least non-transparent.